subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -66,6 +66,15 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(bt_ctlr_hci_driver);
 
+#define DT_DRV_COMPAT zephyr_bt_hci_ll_sw_split
+
+struct hci_driver_data {
+	bt_hci_recv_t recv;
+};
+
+struct hci_driver_config {
+};
+
```

Comment:
Can you add a in-code comment why we have a struct with a single field and another struct with no fields?

Reply:
> Can you add a in-code comment why we have a struct with a single field and another struct with no fields?

The second struct should be removed. I think it's a leftover from when I initially had something in it. The Zephyr driver & device model is such that you can have both read-write and read-only data associated with a `struct device`. The read-only data is found behind the `dev->config` pointer and the read-write behind the `dev->data` pointer. These structs were named accordingly.